source/UI/UITowerShop/UITowerShop.gd

```diff
@@ -35,7 +35,7 @@ func _on_Tower_sold(price: int, _position: Vector2) -> void:
 
 func _on_Player_gold_changed(gold_amount: int) -> void:
 	for button in _buttons:
-		if button.tower_cost > player.gold:
+		if button.tower_cost > gold_amount:
 			button.disabled = true
 		else:
 			button.disabled = false
```

Comment:
I always forget but these are good candidates for Boolean expressions. 

`button.disabled = button.tower_cost > gold_amount`